components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/executor/PasswordProvisioningExecutor.java

```diff
@@ -156,23 +158,19 @@ private String getStringProperty(FlowExecutionContext context, String key) {
     private void handlePrePasswordUpdate(User user, String recoveryScenario, String confirmationCode)
             throws IdentityRecoveryException, IdentityEventException {
 
-        updateIdentityContext();
+        enterFlow();
```

Comment:
Shall we add the try and finally block here and exit the flow, rather than exiting in a different place ?